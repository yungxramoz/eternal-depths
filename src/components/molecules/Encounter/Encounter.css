.encounter-name {
  padding-bottom: 3em;
}

.animate-damaging {
  animation: damage 0.5s;
}

.animate-attacking {
  animation: attack 0.5s;
}

.animate-idle-flying {
  animation: flying 1.2s ease infinite;
}

.animate-idle-standing {
  animation: standing 2s linear infinite;
}

@keyframes damage {
  0% {
    transform: rotate(3deg);
  }
  25% {
    filter: grayscale(1) drop-shadow(-2px 2px 5px black);
    opacity: 0.8;
  }
  50% {
    filter: grayscale(0.3) drop-shadow(-2px 2px 5px black);
  }
}

@keyframes attack {
  0% {
    transform: scale(1.1);
  }
}

@keyframes flying {
  0% {
    transform: translate(0px);
  }
  50% {
    transform: translateY(3px);
    filter: drop-shadow(-2px 2px 4px black);
  }
  100% {
    transform: translate(0px);
  }
}

@keyframes standing {
  50% {
    transform: translateX(2px) rotate(1deg);
  }
}
