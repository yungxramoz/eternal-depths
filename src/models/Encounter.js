import { ENCOUNTERS } from '../constants/encounter'
import { ENCOUNTER_BOSS } from '../constants/encounter-boss'

class Encounter {
  constructor(level, isBoss = false, randomEncounter = true, encounter = null) {
    if (randomEncounter) {
      const encounters = isBoss ? ENCOUNTER_BOSS : ENCOUNTERS
      const randomIndex = Math.floor(Math.random() * encounters.length)
      const randomEncounter = encounters[randomIndex]
      encounter = randomEncounter
    }

    this.level = level
    this.isBoss = isBoss
    this.name = encounter.name
    this.fileName = encounter.fileName
    this.style = encounter.style

    const points = level * 2
    const stats = encounter.baseStats
    for (let i = 0; i < points; i++) {
      const randomStat = Math.floor(Math.random() * 4)
      switch (randomStat) {
        case 0:
          stats.health += 1
          break
        case 1:
          stats.strength += 1
          break
        case 2:
          stats.agility += 1
          break
        case 3:
          stats.precision += 1
          break
        default:
          break
      }
    }
    this.stats = stats
    this.maxHp = stats.health * 3 + 10
    this.hp = this.maxHp
    this.minDamage = stats.strength * 2
    const maxDamageRange = isBoss ? 5 : 3
    this.maxDamage = this.minDamage + maxDamageRange
  }

  attack() {
    return (
      Math.floor(Math.random() * (this.maxDamage - this.minDamage)) +
      this.minDamage
    )
  }

  takeDamage(damage) {
    this.hp -= damage
  }

  isDefeated() {
    return this.hp <= 0
  }
}

export default Encounter
