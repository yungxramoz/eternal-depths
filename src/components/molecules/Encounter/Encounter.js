import React from 'react'
import EncounterImg from '../../atoms/EncounterImg/EncounterImg'
import HpProgressBar from '../ProgressBar/HpProgressBar'
import './Encounter.css'

const Encounter = ({ encounterAnimationState, encounter, showHp }) => {
  return (
    <>
      <h1>
        {encounter.name} Lvl {encounter.level}
      </h1>
      {showHp ? (
        <HpProgressBar currentHp={encounter.hp} maxHp={encounter.maxHp} />
      ) : null}
      <EncounterImg
        encounterAnimationState={encounterAnimationState}
        encounter={encounter}
      />
    </>
  )
}

export default Encounter
