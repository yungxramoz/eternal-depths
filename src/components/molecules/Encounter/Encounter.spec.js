import React from 'react'
import { render } from '@testing-library/react'
import Encounter from './Encounter'

jest.mock('../../atoms/EncounterImg/EncounterImg')
describe('Encounter', () => {
  const encounter = {
    name: 'Test Encounter',
    level: 1,
    hp: 50,
    maxHp: 100,
  }

  it('renders correctly', () => {
    render(<Encounter encounter={encounter} />)
    expect(document.body).toMatchSnapshot('Encounter')
  })
  it('renders correctly with showHp', () => {
    render(<Encounter encounter={encounter} showHp />)
    expect(document.body).toMatchSnapshot('Encounter with showHp')
  })
})
