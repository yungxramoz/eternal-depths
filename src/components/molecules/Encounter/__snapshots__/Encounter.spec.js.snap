// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Encounter renders correctly with showHp: Encounter with showHp 1`] = `
<body>
  <div>
    <h1>
      Test Encounter
       Lvl 
      1
    </h1>
    <div
      class="progress-container"
    >
      <p
        class="hint-text"
      >
         HP 
        50
        /
        100
      </p>
      <div
        class="rpgui-progress"
        data-rpguitype="progress"
      >
        <div
          class=" rpgui-progress-track"
        >
          <div
            class="rpgui-progress-fill red progress-animation"
            style="left: 0px; width: 50%;"
          />
        </div>
        <div
          class=" rpgui-progress-left-edge"
        />
        <div
          class=" rpgui-progress-right-edge"
        />
      </div>
    </div>
  </div>
</body>
`;

exports[`Encounter renders correctly: Encounter 1`] = `
<body>
  <div>
    <h1>
      Test Encounter
       Lvl 
      1
    </h1>
  </div>
</body>
`;
